lstchain/reco/dl1_to_dl2.py

```diff
@@ -51,6 +53,159 @@
+    between the zenith pointings of the MC data in the training sample on
+    which the RFs were trained.
+
+    Parameters:
+    -----------
+    dl2table: pandas dataframe. Four columns will be added: alt0, alt1, w0, w1
+    alt0 and alt1 are the alt_tel values (telescope elevation, in radians) of
+    the closest and second-closest training MC pointings (closest in elevation,
+    on the same side of culmination) for each event in the table. The values
+    w0 and w1 are the corresponding weights that, multiplied by the RF
+    predictions at those two pointings, provide the interpolated result for
+    each event's pointing
+
+    training_zd_deg: array containing the zenith distances (in deg) for the
+    MC training nodes
+
+    training_az_deg: array containing the azimuth angles (in deg) for the
+    MC training nodes (a given index in bith arrays corresponds to a given MC
+    pointing)
+
+    Returns:
+    -------
+    DL2 pandas dataframe with additional columns alt0, alt1, w0, w1
+
+    """
+    pd.options.mode.copy_on_write = True
+
+    alt_tel = dl2table['alt_tel']
+    az_tel  = dl2table['az_tel']
+
+    training_alt_rad = np.pi / 2 - np.deg2rad(training_zd_deg)
+    training_az_rad = np.deg2rad(training_az_deg)
+
+    tiled_az = np.tile(az_tel,
+                       len(training_az_rad)).reshape(len(training_az_rad),
+                                                     len(dl2table)).T
+    tiled_alt = np.tile(alt_tel,
+                       len(training_alt_rad)).reshape(len(training_alt_rad),
+                                                      len(dl2table)).T
+
+    delta_alt = np.abs(training_alt_rad - tiled_alt)
+    # mask to select training nodes only on the same side of the source
+    # culmination as the event:
+    same_side_of_culmination = np.sign(np.sin(training_az_rad) *
+                                       np.sin(tiled_az)) > 0
+    # Just fill a large value for pointings on the other side of culmination:
+    delta_alt = np.where(same_side_of_culmination, delta_alt, np.pi/2)
+    # indices ordered according to distance in telescope elevation
+    sorted_indices = np.argsort(delta_alt, axis=1)
+    closest_alt = training_alt_rad[sorted_indices[:, 0]]
+    second_closest_alt = training_alt_rad[sorted_indices[:, 1]]
+
+    c0 = np.cos(np.pi / 2 - closest_alt)
+    c1 = np.cos(np.pi / 2 - second_closest_alt)
+    cos_tel_zd = np.cos(np.pi / 2 - alt_tel)
+
+    # Compute the weights that multiplied times the RF predictions at the
+    # closest (0) and 2nd-closest (1) nodes (in alt_tel) result in the
+    # interpolated value. Take care of cases in which the two closest nodes
+    # happen to have the same zenith (or very close)! (if so, both nodes are
+    # set to have equal weight in the interpolation)
+    w1 = np.where(np.isclose(closest_alt, second_closest_alt, atol=1e-4, rtol=0),
+                  0.5, (cos_tel_zd - c0) / (c1 - c0))
+    w0 = 1 - w1
+
+    # Update the dataframe:
```

Comment:
I don't see why converting to Series is necessary?

Reply:
Can't recall now the exact error or warning, but I was forced to do this for it to work.

Reply:
I removed the conversion and it still works, thanks!